e2eAIOK/DeNas/search/BaseSearchEngine.py

```diff
@@ -34,18 +37,37 @@ def get_best_structures(self):
     Judge sample structure legal or not
     '''
     def cand_islegal(self, cand):
```

Comment:
please also include your change to cand_isLegal in your workflow design doc.

Reply:
add detailed code flow chart
![image](https://user-images.githubusercontent.com/51695408/218668006-79fd359c-824e-49ec-b92a-df929f1f3fa6.png)